### PR TITLE
Add ability to scroll with the mouse.

### LINKED ISSUE #12: Support for scrolling with mouse
neovim TUI and vanilla `gvim` both support scrolling with the mouse, would be nice to have that in neovim.as too.

### PR DESCRIPTION
**Pull request by rygwdn**
Fixes #12.

### CODE REVIEW

File: src/nvim/ui.coffee [hunk 1]
```diff
@@ -104,6 +106,34 @@ class UI extends EventEmitter
...
+      # Can have both X and Y movement in the same event, so have two separate checks
+      if mouseAbsY > 0
+        direction = if @mouseDeltaY > 0 then 'Up' else 'Down'
+        button = get_vim_button_name('ScrollWheel' + direction, e)
+
+        for i in [1..mouseAbsY]
+          @emit 'input', button
+        @mouseDeltaY = 0
+
+      return false
```
Comment (coolwanglu) [inline src/nvim/ui.coffee]: for consistency, please use `e.preventDefault()` instead of `return false`

File: src/nvim/ui.coffee [hunk 2]
```diff
@@ -104,6 +106,34 @@ class UI extends EventEmitter
...
+
+      mouseAbsX = Math.abs(Math.round(@mouseDeltaX * config.scroll_sensitivity))
+      mouseAbsY = Math.abs(Math.round(@mouseDeltaY * config.scroll_sensitivity))
+
+      if mouseAbsX > 0
+        direction = if @mouseDeltaX > 0 then 'Left' else 'Right'
+        button = get_vim_button_name('ScrollWheel' + direction, e)
+
+        for i in [1..mouseAbsX]
+          @emit 'input', button
```
Comment (coolwanglu) [inline src/nvim/ui.coffee]: Neovim supports scroll event with details on both axes at the same time. Please refer to the gtk-ui in python-client.

**Comment by rygwdn** [comment]
@coolwanglu made both changes as requested. I removed the concept of `scroll_sensitivity`, as I found a more accurate way to calculate the number of rows & columns.

**Comment by rygwdn** [comment]
Note: I haven't managed to get horizontal scrolling to work yet, but I checked that it is sending the right input. I think it might just be something missing in neovim.

**Comment by rygwdn** [comment]
@coolwanglu any feedback on this approach?

**Comment by coolwanglu** [comment]
LGTM.

**Comment by coolwanglu** [comment]
Thanks!

**Pull request merged**